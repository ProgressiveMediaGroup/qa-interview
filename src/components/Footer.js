// Node_Modules
import React, { Component } from 'react'

// Styles
import '../styles/footer.css'

class Footer extends Component {
  render () {
    return (
      <footer>
        <div className='container'>
          &copy; 2017 Progressive Content. All rights reserved.
        </div>
      </footer>
    )
  }
}

export default Footer
