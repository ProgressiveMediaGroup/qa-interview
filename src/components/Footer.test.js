/* global it, describe, expect */
import React from 'react'
import { shallow, mount } from 'enzyme'
import Footer from './Footer'

it('renders without crashing', () => {
  shallow(<Footer />)
})

it('should match snapshot', () => {
  const wrapper = shallow(<Footer />)
  expect(wrapper).toMatchSnapshot()
})

describe('Has correct block structure', function () {
  it('footer tag is displayed', () => {
    const wrapper = mount(<Footer />)
    expect(wrapper.find('footer').length).toBe(1)
  })
  it('container tag is displayed', () => {
    const wrapper = mount(<Footer />)
    expect(wrapper.find('footer').find('div.container').length).toBe(1)
  })
  it('container tag contains correct text', () => {
    const wrapper = mount(<Footer />)
    expect(wrapper.find('footer').find('div.container').text()).toBe('Â© 2017 Progressive Content. All rights reserved.')
  })
})
