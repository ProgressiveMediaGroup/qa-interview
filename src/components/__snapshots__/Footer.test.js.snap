// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        © 2017 Progressive Content. All rights reserved.
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "© 2017 Progressive Content. All rights reserved.",
        "className": "container",
      },
      "ref": null,
      "rendered": "© 2017 Progressive Content. All rights reserved.",
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          © 2017 Progressive Content. All rights reserved.
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 2017 Progressive Content. All rights reserved.",
          "className": "container",
        },
        "ref": null,
        "rendered": "© 2017 Progressive Content. All rights reserved.",
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
